### PR TITLE
fix(grounding): normalize backticked identifiers (#269, #262)

### LINKED ISSUE #262: extract_identifiers misses backticked calls with arguments
## Description

`extract_identifiers` in `src/grounding.rs` (L57-68) only strips trailing literal `()` from backticked identifiers. When an LLM backticks a call expression with arguments like `` `parse_unified_diff(input)` `` or `` `foo(bar, baz)` ``, the identifier is kept verbatim and searched literally, which will not match the function declaration or bare symbol in source.

This causes false `SymbolNotFound` demotions even when the cited symbol exists.

## Related

- #257 (UTF-8 byte filtering in extract_identifiers)
- Previously fixed: trailing `()` stripping in commit e527666

## Suggested Fix

Strip everything from the first `(` onwards when normalizing backticked identifiers, not just a trailing literal `()`.

Found by quorum self-review of #259/#260 branch.

### LINKED ISSUE #269: grounding: extract_identifiers accepts non-symbol backtick text
## Context

Quorum self-review finding on src/grounding.rs:58-71.

`extract_identifiers` treats any backtick span as an identifier after filtering stopwords and short tokens. Prose like \`missing check\` or \`line 42\` passes through and causes spurious `SymbolNotFound` grounding results.

## Proposed fix

Add a `looks_like_symbol` validator that requires the extracted text to contain only identifier-like characters (alphanumeric, `_`, `:`, `.`).

Found by: quorum self-review on PR #268

### PR DESCRIPTION
**Pull request by jsnyder**
## Summary
- Adds `looks_like_symbol` validator that rejects whitespace, non-alpha content, and disallowed characters from backtick captures
- Adds normalization pipeline to `extract_identifiers`: strip generic type params (`<`/`>`), strip call arguments (`(`), clean trailing `::` and `.`
- Fixes ordering: generics stripped before args to handle tuple/fn syntax inside type params (e.g. `Decoder<(u32, u32)>`)
- Restores `collect` to STOPWORDS (caught by independent code review)
- 29 new tests covering happy path, edge cases, rejections, and regression guards

## Quorum verdicts recorded
- **TP** (post_fix): generic/tuple ordering bug — `Option<(u32, u32)>` truncated at `(` before generics could strip
- **FP** (pattern-overgeneralization): prose words suggestion — intentionally accepted by `looks_like_symbol` design

## Test plan
- [x] All 2106 tests pass (`cargo test --bin quorum` + `cargo test --lib`)
- [x] `cargo clippy -- -D warnings` clean
- [x] Quorum self-review: no in-branch findings remaining
- [x] Independent code review pass completed

Closes #269, closes #262

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced identifier extraction to properly handle Rust generics and function calls with improved validation.

* **Tests**
  * Expanded test coverage for identifier validation edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jsnyder/quorum/pull/285)

### CODE REVIEW

File: src/grounding.rs [hunk 1]
```diff
@@ -63,17 +63,52 @@ static STOPWORDS: LazyLock<HashSet<&'static str>> = LazyLock::new(|| {
...
+            if let Some(pos) = id.find('(') {
+                id = id[..pos].trim_end();
+            }
+
+            // Post-strip cleanup: trailing :: and .
+            id = id.trim_end_matches([':', '.']);
+
+            if id.len() >= MIN_IDENTIFIER_LEN
+                && !STOPWORDS.contains(id)
+                && looks_like_symbol(id)
```
[the remark this answers is not in the document]
Comment (jsnyder) [reply src/grounding.rs]: Done in fde3fa2 — added symmetric `trim_start_matches` and a test for `::some_func`.
